solidity/test/bridge/VendingMachine.test.js

```diff
@@ -0,0 +1,604 @@
+        expect(await vendingMachine.unmintFee()).to.equal(unmintFee)
+      })
+
+      it("should start the governance delay timer", async () => {
+        expect(await vendingMachine.getRemainingUnmintFeeUpdateTime()).to.equal(
+          604800 // 7 days contract governance delay
+        )
+      })
+
+      it("should emit UnmintFeeUpdateInitiated event", async () => {
+        await expect(tx)
+          .to.emit(vendingMachine, "UnmintFeeUpdateInitiated")
+          .withArgs(newUnmintFee, await lastBlockTime())
+      })
+    })
+  })
+
+  describe("finalizeUnmintFeeUpdate", () => {
+    context("when caller is not the owner", () => {
+      it("should revert", async () => {
+        await expect(
+          vendingMachine.connect(thirdParty).finalizeUnmintFeeUpdate()
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when caller is the owner", () => {
+      context("when update process is not initialized", () => {
+        it("should revert", async () => {
+          await expect(
+            vendingMachine.connect(governance).finalizeUnmintFeeUpdate()
+          ).to.be.revertedWith("Change not initiated")
+        })
+      })
+
+      context("when update process is initialized", () => {
+        const newUnmintFee = 151511
+
+        beforeEach(async () => {
+          await vendingMachine
+            .connect(unmintFeeUpdateInitiator)
+            .initiateUnmintFeeUpdate(newUnmintFee)
+        })
+
+        context("when governance delay has not passed", () => {
+          it("should revert", async () => {
+            await increaseTime(601200) // +7 days 23 hours
+            await expect(
+              vendingMachine.connect(governance).finalizeUnmintFeeUpdate()
+            ).to.be.revertedWith("Governance delay has not elapsed")
+          })
+        })
+
+        context("when governance delay passed", () => {
+          let tx
+
+          beforeEach(async () => {
+            await increaseTime(604800) // +7 days contract governance delay
+            tx = await vendingMachine
+              .connect(governance)
+              .finalizeUnmintFeeUpdate()
+          })
+
+          it("should update the unmint fee", async () => {
+            expect(await vendingMachine.unmintFee()).to.equal(newUnmintFee)
+          })
+
+          it("should emit UnmintFeeUpdated event", async () => {
+            await expect(tx)
+              .to.emit(vendingMachine, "UnmintFeeUpdated")
+              .withArgs(newUnmintFee)
+          })
+
+          it("should reset the governance delay timer", async () => {
+            await expect(
+              vendingMachine.getRemainingUnmintFeeUpdateTime()
+            ).to.be.revertedWith("Change not initiated")
+          })
+        })
+      })
+    })
+  })
+
+  describe("unmintFeeFor", () => {
+    const unmintAmount = to1e18(2)
+
+    context("when unmint fee is non-zero", async () => {
```

Comment:
Can we add tests for border cases of minimum and maximum amounts that can be passed to the `unmintFeeFor` function?

Reply:
Hmm... Hard to say what is minimum/maximum and unsure what we'd like to achieve with this test. Make sure there is no underflow/overflow? That depends not only on the amount but also on the fee.

One could argue minimum is 0 but in practice, no one wants to unmint 0 tokens, same for 1, 2, 3, and so on (keep in mind we have 1e18 precision).

For the upper bound... in practice, the maximum is limited by fee and the maximum number of BTC one may want to unmint. IMO, the risk of overflowing is negligible in practice.